resources/aws_ec2.go

```diff
@@ -486,6 +489,68 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsEndpoint := obj.snsServer()
```

Comment:
why are we calling this `snsEndpoint` instead of `snsServer` ? (which is also the name of the function)

Reply:
good point. iirc, at first, snsEndpoint was in the struct, so it couldn't share a name with the method. I'll change it.